src/openrct2/object/Object.h

```diff
@@ -180,6 +181,8 @@ struct ObjectEntryDescriptor
         Generation = ObjectGeneration::JSON;
         Identifier = std::string(newIdentifier);
     }
+
+    explicit ObjectEntryDescriptor(const ObjectRepositoryItem& ori);
```

Comment:
All the other constructors are declared and defined in the header, I'd move their definition to the source file in the future.